server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java

```diff
@@ -127,35 +126,81 @@ public FlowTaskResult result() {
      */
     protected abstract void onUpdate();
 
+    protected void updateStatus(TaskStatus status) {
+        this.status = status;
+    }
+
     private void initTaskMonitor() {
         ThreadFactory threadFactory = new TaskThreadFactory(("Task-Monitor-" + context.getJobIdentity().getId()));
         ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor(threadFactory);
         scheduledExecutor.scheduleAtFixedRate(() -> {
-            if (finished) {
+            if (finished || this.status.isTerminated()) {
                 scheduledExecutor.shutdown();
             }
             try {
                 reportTaskResult();
             } catch (Exception e) {
-                log.warn("Update task progress failed, id: {}", context.getJobIdentity().getId(), e);
+                log.warn("Update task info failed, id: {}", context.getJobIdentity().getId(), e);
             }
-        }, 1, 5, TimeUnit.SECONDS);
+        }, 1, REPORT_TASK_INFO_INTERVAL_SECONDS, TimeUnit.SECONDS);
         log.info("Task monitor init success");
     }
 
     private void reportTaskResult() {
         onUpdate();
-        reporter.report(context.getJobIdentity(), status, progress, result);
-        log.info("Task status: {}, progress: {}%, result: {}", status, String.format("%.2f", progress * 100), result);
+        if (this.status == TaskStatus.DONE) {
+            this.progress = 1.0;
+        }
+        reporter.report(buildCurrentResult());
+        log.info("Report task info, id: {}, status: {}, progress: {}%, result: {}", context.getJobIdentity().getId(),
+                status, String.format("%.2f", progress * 100), result);
     }
 
-    private void updateStatus(TaskStatus status) {
-        this.status = status;
+    private void doFinal() {
+        onUpdate();
+        if (this.status == TaskStatus.DONE) {
+            this.progress = 1.0;
+        }
+        log.info("Task finished with status: {}, id: {}, start to report final result",
+                context.getJobIdentity().getId(), status);
+        DefaultTaskResult finalResult = buildCurrentResult();
+        finalResult.setFinished(true);
```

Comment:
finalResult finished is true , but log is not uploaded at this moment, 
we should split two step, first is upload terminal status like done/failed/canceled,  second upload finished  until log upload completed

Reply:
Okay, the `doFinal()` change to 3 steps:
1. report final result (progress, status, resultJson)
2. upload log file to OSS (TODO)
3. report `finished` signal to ODC server